langages/scala/scala-comme-premier-langage.md

```diff
@@ -0,0 +1,252 @@
+
+Un critère d'accessibilité important est le caractère multiplateforme : le novice doit pouvoir facilement utiliser le langage sur le système d'exploitation avec lequel il est le plus à l'aise. [Scala peut être exécuté sur Windows, Mac OS et Linux](https://www.scala-lang.org/download/).
+
+Que le langage fonctionne sur sa machine c'est bien, mais c'est insuffisant ! Il faut que les outils autour du langage suivent. Le développeur, néophyte inclus, a besoin de facilement éditer son code, de l'exécuter, de le déboguer. Scala propose de nombreux outils, dont certains très faciles d'utilisation. Citons notamment le REPL, installé de base avec le langage, qui permet d'exécuter quelques lignes de Scala à la volée. En voici un aperçu :
+
+![REPL Scala en action]({{ site.baseurl }}/assets/images/scala-repl-screenshot.png)
+
+
+Notons que cette interface interactive embarque de la coloration syntaxique et de l'autocomplétion ! Elle permet de facilement tester de petits fragments de code. À l'opposé, un éditeur comme Visual Studio Code couplé avec un outil de build convient pour des scripts et des projets de plusieurs fichier. Il existe plusieurs outils, mais le plus accessible est certainement [Scala CLI](https://scala-cli.virtuslab.org/), qui se veut plus simple que Maven, Gradle, etc.
+
+Il existe aussi [Scastie, un éditeur en ligne](https://scastie.scala-lang.org/) qui permet d'expérimenter avec Scala et de partager facilement son code, sans rien installer.
+
+Enfin, Scala est un [langage plutôt populaire](https://redmonk.com/rstephens/2022/10/20/top20-jun2022/) qui dispose d'une communauté active (notamment sur les plateformes [Scala Users Forum](https://users.scala-lang.org/) et [Discord](https://discord.gg/scala)) ainsi que de nombreuses [ressources en ligne (cours vidéo, livres, articles, etc.)](https://docs.scala-lang.org/learn.html). Non seulement le débutant peut apprendre en autonomie, mais il peut aussi obtenir de l'aide quand il en a besoin.
+
+## "Scalable language"
+
+Le nom "Scala" est la contraction de "Scalable language", que l'on peut traduire par "langage capable de passer à l'échelle". En d'autres termes, Scala se veut aussi bien adapté pour de petits scripts de quelques lignes que pour de gros projets en entreprise. De ce fait, il permet d'enseigner les bases de la programmation mais aussi des concepts avancés, tels que la programmation parallèle ! L'enseignant pourra donc commencer son cours par des exemples simples dans son navigateur ou dans le REPL, puis entraîner les étudiants sur de petits scripts, et pourquoi pas les faire travailler sur un projet plus important à la fin du semestre.
```

Comment:
la traduction de "scalable" ce serait plutôt "évolutif", "passer à l'échelle" ça fait un peu bizarre je trouve

Reply:
extensif / évolutif oui pourquoi pas si ça semble + clair ^^ 
"passer à l'échelle" ne me dérange pas trop, je l'ai entendu dans plusieurs contextes (il est cité ici https://fr.wikipedia.org/wiki/Extensibilité)

Reply:
Attention à bien diffencier `scale (nun)` de `scalable (adj.)` de `scaling (adj.)` de `to scale (verb)`, aussi il faut comprendre il y a plus de nuance en français qu'en anglais, d'où les multiples traductions possibles.

"mettre à l'échelle" c'est pour traduire "to scale" dans le sens de"augmenter la capacité d'un système à absorber la charge de travail à l'échelle de la charge de travail à absorber", ce n'est pas la bonne traduction ici :p

Reply:
Mais est-ce qu' "augmenter / diminuer la taille du système pour s'adapter au travail" ne convient pas aussi pour quelque chose qui s'adapte à de petits comme à de grands projets ? :thinking:

Reply:
votre article, vous décidez ;)

Reply:
Je propose:
> Le nom "Scala" est la contraction de "Scalable language", que l'on peut traduire littéralement par "langage capable de passer à l'échelle" ou, en meilleur français, "langage évolutif"

Reply:
extensible j'aime bien :)